### PR TITLE
(BIDS-2303) fix luck calculation

### PR DESCRIPTION
**Pull request by MauserBitfly**
### <samp>🤖 Generated by Copilot at 91fd16b</samp>

This pull request enhances the `GetValidatorEarnings` function to use concurrency and epochs for faster and more accurate calculations. It also adds a new `db` function `GetFirstActivationEpoch` to support this feature and removes some redundant code.

### CODE REVIEW

File: handlers/common.go [hunk 1]
```diff
@@ -364,7 +344,7 @@ func getProposalLuck(slots []uint64, validatorsCount int) float64 {
 	// Get the timeframe for which we should consider qualified proposals
 	var proposalTimeframe time.Duration
 	// Time since the first block in the proposed block slice
-	timeSinceFirstBlock := time.Since(utils.SlotToTime(slots[0]))
+	timeSinceFirstBlock := time.Since(utils.EpochToTime(fromEpoch))
```
Comment (LuccaBitfly) [inline handlers/common.go:347]: variable name and comment not accurate

**Review by LuccaBitfly — CHANGES_REQUESTED** [review]
works fine, just some small nitpicks

File: db/db.go [hunk 2]
```diff
@@ -3159,6 +3159,17 @@ func GetTotalValidatorDeposits(validators []uint64, totalDeposits *uint64) error
 	`, validatorsPQArray)
 }
 
+func GetFirstActivationEpoch(validators []uint64, totalWithdrawals *uint64) error {
+	validatorsPQArray := pq.Array(validators)
+	return ReaderDb.Get(totalWithdrawals, `
```
[the remark this answers is not in the document]
Comment (MauserBitfly) [reply db/db.go]: thanks, renamed

File: db/db.go [hunk 3]
```diff
@@ -3159,6 +3159,17 @@ func GetTotalValidatorDeposits(validators []uint64, totalDeposits *uint64) error
...
+		SELECT 
+			activationepoch
+		FROM validators
+		WHERE validatorindex = ANY($1) 
+		ORDER BY activationepoch limit(1)
```
[the remark this answers is not in the document]
Comment (MauserBitfly) [reply db/db.go]: corrected to the standard

File: handlers/common.go (hunk 1 above)
Comment (MauserBitfly) [reply handlers/common.go]: renamed and changed comment

**Review by LuccaBitfly — APPROVED** [review]
lgtm

**Review by recy21 — APPROVED** [review]
lgtm, index present for validatorindex & activationepoch :+1:

**Pull request merged**